docs/reference/builder.md

```diff
@@ -1174,8 +1174,9 @@ To use these, simply pass them on the command line using the `--build-arg
 `ARG` variables are not persisted into the built image as `ENV` variables are.
 However, `ARG` variables do impact the build cache in similar ways. If a
 Dockerfile defines an `ARG` variable whose value is different from a previous
-build, then a "cache miss" occurs upon first use of the `ARG` variable. The
-declaration of the `ARG` variable does not count as a use.
+build, then a "cache miss" occurs upon its first usage, not its declaration. In
+particular, all `RUN` following the `ARG` definition do use the `ARG` variable
```

Comment:
`definition` or `declaration` ? We use the term `declaration` in the previous sentence. It's the only use of `declaration`. Maybe it could be changed to `definition` too.

Also, is `directive` a standard word in `Dockerfile` vocabulary ? In the intro they are called `instructions`.

Words count:
* `definition`: 4, always used to refer to the variable defined by the `ARG` instruction
* `defin*`: 12
* `declaration`: 0
* `instruction`: 95
* `statement`: 3, they could maybe be changed to `instruction` too

I pushed a new commit on this PR to improve this.